dataactvalidator/validation_handlers/file_generation_manager.py

```diff
@@ -27,129 +20,92 @@ class FileGenerationManager:
+        if self.file_generation:
+            self.generate_d_file(file_path)
+        elif self.file_type == 'E':
+            self.generate_e_file()
+        else:
+            self.generate_f_file()
+
+        log_data = {
+            'message': 'Finished file {} generation'.format(self.file_type), 'message_type': 'ValidatorInfo',
+            'file_type': self.file_type, 'file_path': file_path
+        }
+        if self.file_generation:
+            log_data.update({
+                'agency_code': self.file_generation.agency_code, 'agency_type': self.file_generation.agency_type,
+                'start_date': self.file_generation.start_date, 'end_date': self.file_generation.end_date,
+                'file_generation_id': self.file_generation.file_generation_id
+            })
+        else:
+            log_data['job_id'] = self.job.job_id
         logger.info(log_data)
 
-        # Get or create a FileRequest for this generation
-        current_date = datetime.now().date()
-        file_request_params = {
-            "job_id": self.job.job_id, "is_cached_file": True, "start_date": self.job.start_date,
-            "end_date": self.job.end_date, "agency_code": self.agency_code, "file_type": self.job.file_type.letter_name,
-            "agency_type": self.agency_type
+    def generate_d_file(self, file_path):
+        """ Write file D1 or D2 to an appropriate CSV. """
+        log_data = {
+            'message': 'Starting file {} generation'.format(self.file_type), 'message_type': 'ValidatorInfo',
+            'agency_code': self.file_generation.agency_code, 'agency_type': self.file_generation.agency_type,
+            'start_date': self.file_generation.start_date, 'end_date': self.file_generation.end_date,
+            'file_generation_id': self.file_generation.file_generation_id, 'file_type': self.file_type,
+            'file_path': file_path
         }
+        logger.info(log_data)
 
-        file_request = self.sess.query(FileRequest).filter_by(**file_request_params).one_or_none()
-        if not file_request:
-            file_request_params["request_date"] = current_date
-            file_request = FileRequest(**file_request_params)
-            self.sess.add(file_request)
-            self.sess.commit()
-
-        # Mark this Job as not from-cache, and mark the FileRequest as the cached version (requested today)
-        self.job.from_cached = False
-        file_request.is_cached_file = True
-        file_request.request_date = current_date
-        self.sess.commit()
+        original_filename = file_path.split('/')[-1]
+        local_file = "".join([CONFIG_BROKER['d_file_storage_path'], original_filename])
 
         # Prepare file data
-        file_utils = fileD1 if self.job.file_type.letter_name == 'D1' else fileD2
-        local_file = "".join([CONFIG_BROKER['d_file_storage_path'], self.job.original_filename])
+        file_utils = fileD1 if self.file_type == 'D1' else fileD2
```

Comment:
Again, better to be explicit and force the `file_type` to _be_ `'D2'` before falling-back to D2.